### ISSUE TITLE
v0.3.0 - Driver failed to start

### ISSUE THREAD

**Issue by kkellner** [hassio-addons/addon-nut#49]
# Problem/Motivation

Upgraded to v0.3.0 of nut addon and driver fails to start

## Expected behavior

Backward compatibility with prior version of plugin

## Actual behavior

Startup failed with the following log output:

```
-----------------------------------------------------------
 Add-on: Network UPS Tools
 Manage battery backup (UPS) devices
-----------------------------------------------------------
 Add-on version: 0.3.0
 You are running the latest version of this add-on.
 System: HassOS 4.12  (amd64 / qemux86-64)
 Home Assistant Core: 0.114.4
 Home Assistant Supervisor: 235
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
Log level is set to DEBUG
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] nut.sh: executing... 
[22:29:23] INFO: Setting mode to netserver...
[22:29:23] INFO: Generating /etc/nut/upsd.users...
[22:29:23] INFO: Configuring user: nutuser
[22:29:24] DEBUG: Password SHA1: xxxxxxx
[22:29:24] DEBUG: Requested API resource: https://api.pwnedpasswords.com/range/52817
[22:29:24] DEBUG: API HTTP Response code: 200
[22:29:24] INFO: Password is NOT in the Have I Been Pwned database! Nice!
[22:29:24] INFO: Configuring Device named apcups...
[22:29:24] INFO: Starting the UPS drivers...
Network UPS Tools - UPS driver controller 2.7.4
   0.000000	Starting UPS: apcups
Error loading shared library libnetsnmp.so.35: No such file or directory (needed by /usr/lib/nut/snmp-ups)
Error relocating /usr/lib/nut/snmp-ups: snmp_open: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_api_errstring: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snprint_objid: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_clone_pdu: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_add_var: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_sess_init: symbol not found
Error relocating /usr/lib/nut/snmp-ups: netsnmp_oid_equals: symbol not found
Error relocating /usr/lib/nut/snmp-ups: init_snmp: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_parse_oid: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_free_pdu: symbol not found
Error relocating /usr/lib/nut/snmp-ups: generate_Ku: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_error: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_close: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_errstring: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_out_toggle_options: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_pdu_create: symbol not found
Error relocating /usr/lib/nut/snmp-ups: read_objid: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_sess_error: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snprint_hexstring: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_synch_response: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_add_null_var: symbol not found
Error relocating /usr/lib/nut/snmp-ups: snmp_errno: symbol not found
Error relocating /usr/lib/nut/snmp-ups: usmAES128PrivProtocol: symbol not found
Error relocating /usr/lib/nut/snmp-ups: usmDESPrivProtocol: symbol not found
Error relocating /usr/lib/nut/snmp-ups: usmHMACMD5AuthProtocol: symbol not found
Error relocating /usr/lib/nut/snmp-ups: usmHMACSHA1AuthProtocol: symbol not found
   0.004556	Driver failed to start (exit status=127)
[cont-init.d] nut.sh: exited 1.
[cont-finish.d] executing container finish scripts...
[cont-finish.d] 99-message.sh: executing... 
-----------------------------------------------------------
                Oops! Something went wrong.

 We are so sorry, but something went terribly wrong when
 starting or running this add-on.
 
 Be sure to check the log above, line by line, for hints.
-----------------------------------------------------------
[cont-finish.d] 99-message.sh: exited 0.
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
```

## Steps to reproduce

Using the following configuration:
```
users:
  - username: '!secret ups_tools_username'
    password: '!secret ups_tools_password'
    instcmds:
      - all
    actions: []
devices:
  - name: apcups
    driver: snmp-ups
    port: apc01.mydomain.com
    config:
      - snmp_version = v3
      - secLevel = authPriv
      - secName = snmpprofile1
      - authPassword = xxxxx
      - authProtocol = MD5
      - privPassword = xxxxx
      - privProtocol = DES
      - desc = APC UPS
mode: netserver
shutdown_host: 'false'
log_level: debug
```

**Comment by tnagels** [comment]
I have the same problem, nut won't start anymore. My log looks almost exactly the same, the problem seems to be:
`Error loading shared library libnetsnmp.so.35: No such file or directory (needed by /usr/lib/nut/snmp-ups)`

Thomas

**Comment by sinclairpaul** [comment]
If you can please test the latest version on the Edge repo (as I have added the missing libraries).  I have some concerns if SNMPv3 may or may not work.

**Comment by kkellner** [comment]
I will need instructions on how to load/install an alternative version of the addon.

**Comment by sinclairpaul** [comment]
Add the repo (Supervisor/Add-on Store) three dots, repositories

Paste in `https://github.com/hassio-addons/repository-edge`

You should then be able to install from the edge repo (scroll down to see the addons in the list).  Please note any sensors in HA itself may or may not work based on how they are configured.

**Comment by kkellner** [comment]
Well the addon started, but the sensors were not working, so I deleted the integration and tried to add it back but it is unable to connect.   Is the addon hostname still `a0d7b954-nut` when running from "Edge" repo?  It's been awhile since I configured this, maybe something else I missing?

**Comment by sinclairpaul** [comment]
The sensors not being recognised is expected (the addon has a different internal docker name).  Can you post the startup log so I can validate, then I will push an update.

**Comment by sinclairpaul** [comment]
Just to add, if you really want to validate the name should be `40817795-nut`

**Comment by kkellner** [comment]
Successful startup (or at least it doesn't terminate):

```
[18:55:47] INFO: Configuring user: nutuser
[18:55:48] DEBUG: Password SHA1: xxxxx
[18:55:48] DEBUG: Requested API resource: https://api.pwnedpasswords.com/range/52817
[18:55:48] DEBUG: API HTTP Response code: 200
[18:55:48] INFO: Password is NOT in the Have I Been Pwned database! Nice!
[18:55:48] INFO: Configuring Device named apcups...
[18:55:48] INFO: Starting the UPS drivers...
Network UPS Tools - UPS driver controller 2.7.4
   0.000000	Starting UPS: apcups
Network UPS Tools - Generic SNMP UPS driver 0.97 (2.7.4)
Created directory: /var/lib/net-snmp
Created directory: /var/lib/net-snmp/mib_indexes
No matching MIB found for sysOID '.1.3.6.1.4.1.318.1.3.27'!
Please report it to NUT developers, with an 'upsc' output for your device.
Going back to the classic MIB detection method.
Detected Smart-UPS 1500 on host apc01.local (mib: apcc 1.2)
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.3.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.3.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.3.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.4.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.4.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.4.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.5.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.5.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.5.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.6.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.6.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.6.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.7.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.7.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.7.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.8.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.8.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.8.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.2.1.4.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.2.2.7.0
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.2.3.5.0
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.2.2.9.0
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.2.3.6.0
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.2.2.5.0
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.2.2.6.0
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.2.1.4.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.3.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.3.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.3.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.4.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.4.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.4.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.5.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.5.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.5.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.6.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.6.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.6.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.7.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.7.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.7.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.8.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.8.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.8.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.9.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.9.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.9.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.10.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.10.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.10.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.11.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.11.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.11.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.12.1.1.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.12.1.1.2
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.3.3.1.12.1.1.3
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.2.1.1.0
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.10.1.2.2.1.3.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.10.1.2.2.1.4.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.2.1.2.0
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.10.1.2.2.1.6.1
[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.10.1.2.2.1.7.1
[cont-init.d] nut.sh: exited 0.
[cont-init.d] nutclient.sh: executing... 
[cont-init.d] nutclient.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[18:55:52] INFO: Starting the UPS information server...
Network UPS Tools upsd 2.7.4
   0.000000	fopen /var/run/nut/upsd.pid: No such file or directory
   0.000321	listening on 0.0.0.0 port 3493
   0.000596	Connected to UPS [apcups]: snmp-ups-apcups
   0.001241	/etc/nut/upsd.users is world readable
[18:55:52] INFO: Starting the UPS monitor and shutdown controller...
Network UPS Tools upsmon 2.7.4
   0.000000	fopen /var/run/upsmon.pid: No such file or directory
   0.000157	Using power down flag file /etc/killpower
   0.000315	UPS: apcups@localhost (master) (power value 1)
   0.000372	debug level is '1'
   0.000376	Warning: running as one big root process by request (upsmon -p)
   0.000538	Trying to connect to UPS [apcups@localhost]
   0.007873	User upsmonmaster@127.0.0.1 logged into UPS [apcups]
   0.001040	Logged into UPS apcups@localhost
```

**Comment by sinclairpaul** [comment]
Did you receive the same messages previously from the SNMP mib i.e. `[apcups] unhandled ASN 0x81 received from .1.3.6.1.4.1.318.1.1.1.9.2.3.1.3.1.1.1`?

**Comment by kkellner** [comment]
I do not recall if that message was there before.   As far as I can tell I have the sensors I expect.

**Comment by sinclairpaul** [comment]
Thanks, I will push the release.

**Comment by sinclairpaul** [comment]
Should be fixed with v0.3.1

**Comment by sinclairpaul** [comment]
Thanks for the help @kkellner 👍

**Comment by kkellner** [comment]
Switched from Edge to v3.1.   Working fine.  Thanks for the fast fix!

**Comment by lelepando** [comment]
Fixed with 0.3.1! thanks a lot!!!
